src/ports.js

```diff
@@ -418,6 +418,25 @@
             return this;
         },
 
+        removePorts: function(ports, opt) {
+
+            var options = opt || {};
+
+            if (ports.length) {
+                options.rewrite = true;
+                var remainingPorts = util.assign([], this.prop('ports/items'));
+                ports.forEach(function(port) {
+                    var id = util.isObject(port) ? port.id : port;
+                    remainingPorts = remainingPorts.filter(function(rPort) {
```

Comment:
wouldn't be even faster if the `filter` weren't here, in every iteration?

Reply:
Do you have psudocode for another way to do it?

I wasn't able to find any better ways to do "array subtraction" in javascript

Reply:
I would prefer to filter the ports once, but search multiple times.
```javascript
var remainingPorts = currentPorts.filter(cp => !portsForRemoval.some(rp => portsEquals(cp, rp)));
```

Reply:
Thanks, yeah that worked very cleanly.